internal/internal_event_handlers.go

```diff
@@ -1067,38 +1077,48 @@ func (weh *workflowExecutionEventHandlerImpl) handleMarkerRecorded(
 	eventID int64,
 	attributes *eventpb.MarkerRecordedEventAttributes,
 ) error {
-	encodedValues := newEncodedValues(attributes.Details, weh.dataConverter)
 	switch attributes.GetMarkerName() {
 	case sideEffectMarkerName:
-		var sideEffectID int64
-		var result *commonpb.Payloads
-		_ = encodedValues.Get(&sideEffectID, result)
-		weh.sideEffectResult[sideEffectID] = result
+		for sideEffectID, result := range attributes.GetDetails() {
+			weh.sideEffectResult[sideEffectID] = result
+		}
```

Comment:
Before only one `sideEffectID` (and all other markers bellow) was supported. Now it copies all of them from `Details`. I can add `break` here to have only first item copied. There should be only one anyway.

Reply:
The cleaner approach would be to have fixed keys: "result", "id" and put payload and id into values.